rlp/decode.go

```diff
@@ -697,6 +709,24 @@ func (s *Stream) uint(maxbits int) (uint64, error) {
 	}
 }
 
+// Bool reads an RLP string of up to 1 byte and returns its contents
+// as an boolean. If the input does not contain an RLP string, the
+// returned error will be ErrExpectedString.
+func (s *Stream) Bool() (bool, error) {
+	num, err := s.uint(8)
+	if err != nil {
+		return false, err
+	}
+	switch num {
+	case 0:
+		return false, nil
+	case 1:
+		return true, nil
+	default:
+		return false, errBoolInvalid
```

Comment:
Can you put the value of `num` into the message again? If someone gives you `2` and it fails,
it's going to be helpful to see that immediately.